src/ra_server_proc.erl

```diff
@@ -1188,6 +1230,81 @@ handle_effect(RaftState, {aux, Cmd}, EventType, State0, Actions0) ->
+                                                      ra:remove_member(Members, ServerId),
+                                                      {error, timeout};
+                                                  E ->
+                                                      ra:force_delete_server(System, ServerId),
+                                                      E
+                                              end;
+                                          E ->
+                                              E
+                                      end,
+                                ok = gen_statem:cast(Me, {add_member, ServerId, Res})
+                        end)
+          end,
+    %% Q: Storing the Pid, but is monitoring needed, or just check if
+    %% process is alive in each lap?
+    {State#state{monitors = ra_monitors:add(Pid, member_eval, Monitors),
+                 server_state = SS#{member_eval_pid => Pid}}, Actions};
+handle_effect(_RaftState, {remove_member, ServerId, Members}, _EventType,
+              #state{server_state = SS, monitors = Monitors} = State, Actions) ->
+    #{name := System} = ra_server:system_config(SS),
+    #{member_eval_pid := OldPid} = SS,
+    Me = self(),
+    %% Check if an member_eval pid is already ongoing. Do we need to monitor the pid?
+    Pid = case is_pid(OldPid) andalso is_process_alive(OldPid) of
+              true ->
+                  %% There is already a process working, do nothing
+                  OldPid;
+              false ->
+                  %% No process working, safe to spawn a new one.
+                  spawn(fun() ->
+                                Res = case ra:remove_member(Members, ServerId) of
+                                          {ok, _, _Leader} = R ->
+                                              case ra:force_delete_server(System, ServerId) of
+                                                  ok ->
+                                                      R;
+                                                  {error, {badrpc, nodedown}} ->
+                                                      R;
+                                                  {error, {badrpc, {'EXIT', {badarg, _}}}} ->
+                                                      R;
+                                                  {error, Err} = Err ->
+                                                      {error, Err, R};
+                                                  Err ->
+                                                      {error, Err, R}
+                                              end;
+                                          {timeout, _} ->
+                                              {error, timeout};
+                                          E ->
+                                              E
+                                      end,
+                                ok = gen_statem:cast(Me, {remove_member, ServerId, Res})
```

Comment:
I am thinking we can remove all the result handling and just reset the timer on monitor down.

Reply:
Done.